src/ShaderGen/ShaderMethodVisitor.cs

```diff
@@ -211,6 +211,19 @@ public override string VisitInvocationExpression(InvocationExpressionSyntax node
                     ValidateBuiltInMethod(method);
```

Comment:
Technically, these comparison functions also must be used within a fragment shader. `ValidateBuiltInMethod` basically just does that validation for `Ddx` and `Ddy`, so we can add the logic in there. And perhaps we can make it more general?